src/directives/show.js

```diff
@@ -24,12 +24,14 @@ export function handleShowDirective(component, el, value, modifiers, initialUpda
 
     const handle = (resolve) => {
         if (value) {
-            transitionIn(el,() => {
-                show()
-            }, component)
+            if(el.style.display === 'none' || el.__x_transition) {
```

Comment:
This line removed in v2.4.1 caused a regression experienced by several users.
We need to queue the animation only if the element is hidden or there is a transition running.
We don't check the type of transition and we don't add additional logic here, the low level functions will take care of ensuring that duplicated transition are discarded